org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardSelectObjectsAndPull.java

```diff
@@ -0,0 +1,255 @@
+			String transportRequest) {
+		this.destination = AdtProjectServiceFactory.createProjectService().getDestinationId(project);
+		this.cloneData = cloneData;
+		this.repoToSelectedObjectsMap = new HashMap<String, IAbapGitPullModifiedObjects>();
+		this.transportRequest = transportRequest;
+		fetchModifiedObjects(); //fetch the modified objects for repository and its dependencies
+
+		setWindowTitle(Messages.AbapGitWizardPull_title);
+		setNeedsProgressMonitor(true);
+		setDefaultPageImageDescriptor(
+				AbstractUIPlugin.imageDescriptorFromPlugin(AbapGitUIPlugin.PLUGIN_ID, "icons/wizban/abapGit_import_wizban.png")); //$NON-NLS-1$
+	}
+
+	private void fetchModifiedObjects() {
+		IRepositoryService repoService = RepositoryServiceFactory.createRepositoryService(this.destination, new NullProgressMonitor());
+		IRepositories repositories = repoService.getRepositories(new NullProgressMonitor());
+
+		IAbapGitPullModifiedObjects abapPullModifiedObjects = repoService.getModifiedObjects(new NullProgressMonitor(),
+				repoService.getRepositoryByURL(repositories, this.cloneData.url),
+				this.cloneData.user, this.cloneData.pass);
+
+		if (!abapPullModifiedObjects.getOverwriteObjects().getAbapgitobjects().isEmpty()) {
+			this.cloneData.repoToModifiedOverwriteObjects.add(new ModifiedObjectsForRepository(this.cloneData.url,
+					new ArrayList<IAbapGitObject>(abapPullModifiedObjects.getOverwriteObjects().getAbapgitobjects())));
+		}
+
+		if (!abapPullModifiedObjects.getPackageWarningObjects().getAbapgitobjects().isEmpty()) {
+			this.cloneData.repoToModifiedPackageWarningObjects.add(new ModifiedObjectsForRepository(this.cloneData.url,
+					new ArrayList<IAbapGitObject>(abapPullModifiedObjects.getPackageWarningObjects().getAbapgitobjects())));
+		}
+
+		if (this.cloneData.hasDependencies()) {
+			for (IApackDependency apackDependency : this.cloneData.apackManifest.getDescriptor().getDependencies()) {
+				IRepository dependencyRepository = repoService.getRepositoryByURL(repositories, apackDependency.getGitUrl());
+				if (dependencyRepository != null) {
+					abapPullModifiedObjects = repoService.getModifiedObjects(new NullProgressMonitor(),
+							dependencyRepository, this.cloneData.user,
+							this.cloneData.pass);
+
+					if (!abapPullModifiedObjects.getOverwriteObjects().getAbapgitobjects().isEmpty()) {
+						this.cloneData.repoToModifiedOverwriteObjects.add(new ModifiedObjectsForRepository(dependencyRepository.getUrl(),
+								new ArrayList<IAbapGitObject>(abapPullModifiedObjects.getOverwriteObjects().getAbapgitobjects())));
+					}
+
+					if (!abapPullModifiedObjects.getPackageWarningObjects().getAbapgitobjects().isEmpty()) {
+						this.cloneData.repoToModifiedPackageWarningObjects.add(new ModifiedObjectsForRepository(
+								dependencyRepository.getUrl(),
+								new ArrayList<IAbapGitObject>(abapPullModifiedObjects.getPackageWarningObjects().getAbapgitobjects())));
+					}
+
```

Comment:
refactor the method. we can have a simple method which can be reused for the main repository and as well for dependencies for filling this.cloneData to avoid code duplication.

Reply:
Added method `fetchAndExtractModifiedObjectsToPull` in class `RepositoryUtil` to handle common logic for main repository and dependencies, which can also be used across ui plugin.